src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,53 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final String DATE_FORMAT = "dd.MM.yyyy";
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (names == null || names.length == 0 || data == null || data.length == 0
+                || dateFrom == null || dateFrom == "" || dateTo == null || dateTo == "") {
+            return "";
+        }
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period " + dateFrom + " - " + dateTo);
+        int sumUserSalary = 0;
```

Comment:
can we declare this variable inside a loop?

Reply:
yes, done